File: UserRegistry/src/main/java/com/example/userregistry/services/UserService.java
package com.example.userregistry.services;

import com.example.userregistry.convertor.UserConvertor;
import com.example.userregistry.models.User;
import com.example.userregistry.models.UserDTO;
import com.example.userregistry.repository.UserRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService {

    public static final Logger log = LoggerFactory.getLogger(UserService.class);
    public final UserRepository userRepository;
    public final UserConvertor userConvertor;

    @Autowired
    public UserService(UserRepository userRepository, UserConvertor userConvertor){
        this.userRepository = userRepository;
        this.userConvertor = userConvertor;
    }

    public void saveUser(UserDTO userDTO){
        User userTemp = userConvertor.DTOtoEntity(userDTO);
        userRepository.findUserByPhoneNumber(userDTO.getPhoneNumber()).ifPresent(user ->  userTemp.setId(user.getId()));
        userRepository.save(userTemp);
    }
    public UserDTO getUser(String phoneNumber){
        return userConvertor.entityToDTO(userRepository.findUserByPhoneNumber(phoneNumber).orElse(userRepository.save(new User(phoneNumber))));
    }
}

File: react-client/src/services/user.service.js
import axios from 'axios';
import authHeader from './auth-header';
import {User} from "../models/User";
const API_URL = 'http://localhost:9091/api/v1/user/';
class UserService {
    getUser() {
        return axios.get(API_URL + 'get', { headers: authHeader() });
    }
    saveUser(user: User){
        return axios.post(API_URL+ 'save', user, {headers: authHeader()});
    }

}
export default new UserService();